README.md

```diff
@@ -31,6 +31,25 @@ Content here
 
 Assuming this file is now going to be served from `blog-posts/a-simple-title.html`, a file will be written to `blog-posts/2021/07/21/a-sample-title.html` that redirects to the new URL.s
 
+Bulk Usage
+----------
+
+To perform a page-wide redirect, you can use the `PAST_ARTICLE_URL_FORMATS` setting, which allows you to configure an original URL using page metadata. The structure of this configuration key looks like so:
+
+``` python
+PAST_ARTICLE_URL_FORMATS = {
+    "format-id": {
+        "ARTICLE_URL": "old/layout/{slug}.html",
+        "PAGE_URL": "old/pages/{slug}.html",
+    }
+}
```

Comment:
I chose this name because that's exactly what I'm trying to get at; the idea here is that, if I change my article URL format (say, I used to have `articles/{year}/{month}/{slug}.html') then, I just add those formats to this list, and a full set of redirects will be generated for my _old_ URLs, pointed at the new ones that are defined by the current configuration.

I think both of your questions suggest I haven't documented the use case well enough. Lemme think if I can write it up more clearly.

Reply:
My thinking here was that having redundant redirects was okay, in that at worst we'd have additional names to reach the canonical address. There's also no chain of redirects here; all of them direct only to the canonical content, not in a sequence of redirections.

In case of a mistake in configuration, I'm not sure what to do about that; I don't think that this can suss out the intentions of the site owner :)

Reply:
> My thinking here was that having redundant redirects was okay, in that at worst we'd have additional names to reach the canonical address.

Hmmm... I suppose that's fine. Not strictly correct, but not really hurting anything either 😄 At some point, for large enough sites, this will probably slow things down significantly, but that's an optimization we can address later, hopefully.

> There's also no chain of redirects here; all of them direct only to the canonical content, not in a sequence of redirections.

I think I wasn't clear; I agree there shouldn't be chains of redirection (browsers don't like that, in general).

Then I guess my comments boil down to the following:

1. I think the name of the configuration variable should be changed. This affects more than Articles (in the Pelican context), and it's more general than _past_ articles, although that's certainly the most common case I think. I think `CONTENT_REDIRECT_CONFIG` get's the more general use across.
2. I don't like dictionaries where the keys aren't used for anything. This could be a list/tuple of dictionaries and it would function identically, right? Let's go with one of those then.

Reply:
Oh, I see. Yeah, I had put the keys in there because I was considering adding meta tags to the generated template, but a list of tuples will work just as well, and be extensible. Gimme a bit, I'll amend.

Reply:
I want to keep the "past formats" naming, or at least conceptually. How does "PAST_SITE_LAYOUTS" sound?

Reply:
> but a list of tuples will work just as well

A list of dicts would be better, IMO.

> PAST_SITE_LAYOUTS

The problem I see with "past" is that this is a general redirect functionality. It happens to be useful if you change your site layout and want to redirect from old to new, but you could also have redirects for SEO, say. Why do you prefer the word "past" in there?

Reply:
Oh, okay, that makes sense. I was thinking of it entirely from the use case that "original_url" implied -- "this content used to be there, but it's over here now". I don't really do any SEO, so it wasn't on my radar as a thing that might be needed.

If you prefer the latter case, that's fine. Updating.

Reply:
Yeah, that was my original conception of the idea, but your PR here caused me to realize that it could be actually much more useful than I imagined 😄 So thanks for that!!!